docs/connector-development/config-based/overview.md

```diff
@@ -0,0 +1,95 @@
+
+## Data retriever
+
+The data retriever defines how to read the data from an API source, and acts as an orchestrator for the data retrieval flow.
+There is currently only one implementation, the `SimpleRetriever`, which is defined by
+
+1. Requester: describes how to submit requests to the API source
+2. Paginator <sup>1</sup>: describes how to navigate through the API's pages
+3. Record selector: describes how to select records from an HTTP response
+4. Stream Slicer: describes how to partition the stream, enabling incremental syncs and checkpointing
+
+Each of those components (and their subcomponents) are defined by an explicit interface and one or many implementations.
+The developer can choose and configure the implementation they need depending on specifications of the integrations they are building against.
+
+### Data flow
+
+The retriever acts as a coordinator, moving the data between its components before emitting `AirbyteMessage`s that can be read by the platform.
+The `SimpleRetriever`'s data flow can be described as follows:
+
+1. Given the connection config and the current stream state, the `StreamSlicer` computes the stream slices to read.
+2. Iterate over all the stream slices defined by the stream slicer.
+3. For each stream slice,
+    1. Submit a request as defined by the requester
+    2. Select the records from the response
+    3. Repeat for as long as the paginator points to a next page
+
+More details on the record selector can be found in the [record selector section](record-selector.md)
+More details on the stream slicers can be found in the [stream slicers section](stream-slicers.md)
+More details on the paginator can be found in the [pagination section](pagination.md)
+
+## Requester
+
+The `Requester` defines how to prepare HTTP requests to send to the source API <sup>2</sup>.
+There currently is only one implementation, the `HttpRequester`, which is defined by
+
+1. A base url: the root of the API source
+2. A path: the specific endpoint to fetch data from for a resource
+3. The HTTP method: the HTTP method to use (GET or POST)
+4. A request options provider: defines the request parameters and headers to set on outgoing HTTP requests
+5. An authenticator: defines how to authenticate to the source
+6. An error handler: defines how to handle errors
+
+More details on authentication can be found in the [authentication section](authentication.md).
+More details on error handling can be found in the [error handling section](error-handling.md)
+
+## Connection Checker
+
+The `ConnectionChecker` defines how to test the connection to the integration.
+
+The only implementation as of now is `CheckStream`, which tries to read a record from a specified list of streams and fails if no records could be read.
+
+# Footnotes
+
+1. The paginator is conceptually more related to the requester than the data retriever, but is part of the `SimpleRetriever` because it inherits from `HttpStream` to increase code reusability.
```

Comment:
if you believed it is more related to requester, what is the cost of putting it in the right place?